### ISSUE TITLE
with the new rqlite 6.2.0 is there a new go client for it ? 

### ISSUE THREAD

**Issue by Dorsosiv** [rqlite/rqlite#871]
Hi @otoolep. When I tried to use the rqlite 6.2.0 our microservices did not work well with it as they worked with the old rqlite 6.0.1 version. 
I want to know if there is a new version for the go client that I need to update for our microservices? 

Thanks in advance :)

**Comment by otoolep** [comment]
I would very much like to create a new client, but need to find the time. :-)

What problems are you seeing with the client and 6.0.2? It should work, since the API hasn't changed for rqlite. It could be a bug in 6.0.2. Please let me know.

**Comment by otoolep** [comment]
Sorry, I mean what issues are you seeing with 6.2.0?

**Comment by Dorsosiv** [comment]
Ill deep dive and then give you more logs. What Im seeing , is that the data that we keep in the rqlite is not showed and then is showed and it has a lot of lags. 

Ill check deeper to make sure that the client is or is not the reason.

**Comment by otoolep** [comment]
I changed connection handling significantly in 6.1.0 (so you would also see the changes in 6.2.0 of course) -- the earlier versions actually had a bug and performance issues. From 6.1.0 changes happen on one SQLite connection, and queries happen on another. It is possible you are seeing differences now, and I doubt it's the client. If you're seeing changes it's probably rqlite. 

With rqlite 6.1.0, there are different database connections -- one for changes (execute) and one for queries. So the following applies: https://www.sqlite.org/isolation.html. Perhaps you're seeing the following?

> If the same database is being read and written using two different database connections (two different sqlite3 objects returned by separate calls to sqlite3_open()) and the two database connections do not have a shared cache, then the reader is only able to see complete committed transactions from the writer. Partial changes by the writer that have not been committed are invisible to the reader. This is true regardless of whether the two database connections are in the same thread, in different threads of the same process, or in different processes. This is the usual and expected behavior for SQL database systems.

and
> Changes made in one database connection are invisible to all other database connections prior to commit.

So basically if you are INSERTing using transactions (`transaction` in the URL, which your client might be doing) any queries of rqlite will not see the data until you COMMIT the transaction. It sounds like that's what might be happening. In 6.0.2 both executes and queries happened on the same database connection.

**Comment by Dorsosiv** [comment]
ok Ill check that too, thanks

**Comment by otoolep** [comment]
By the way, the bug is fixed in 6.0.2, but query performance is much better in later versions such as 6.1.0, and 6.2.0. So I guess it depends on what you need right now.

6.0.2: INSERTs and SELECTS cannot happen concurrently. You may not see the "lags" you mentioned.
6.1.0 and later: INSERTs and SELECT can happen concurrently, but you may see "lags".

**Comment by Dorsosiv** [comment]
I think for now we will stick with the 6.0.2 version. Ill save this issue for our developers to see if this change is a hard implementation for the backend microservices

**Comment by otoolep** [comment]
Yes, I'm very interested in learning more about what you're seeing with the later versions. Please let me know more details when you can.
